core/fed_avg_optimizer.py

```diff
@@ -78,8 +78,32 @@ def kickoff(self):
 		Creates a job that transforms and splits the dataset, and also
 		randomly initializes the model.
 		"""
-		self.job.job_type = JobTypes.JOB_INIT.name
-		return ActionableEventTypes.SCHEDULE_JOB.name, self.job
+		job_arr = []
+		job_one = DMLJob(
+			job_type=JobTypes.JOB_INIT.name,
+			serialized_model=self.job.serialized_model,
+			framework_type=self.job.framework_type,
+			weights=self.job.weights,
+			hyperparams=self.job.hyperparams,
+			label_column_name=self.job.label_column_name,
+			raw_filepath=self.job.raw_filepath,
+			session_filepath=self.job.session_filepath,
+			datapoint_count=self.job.datapoint_count
+		)
+		job_arr.append(job_one)
+		job_two = DMLJob(
```

Comment:
Two things.

1. It seems like you're assigning all of the attributes for both the INIT and TRANSFORM_SPLIT job. Why? For INIT, you shouldn't need the session_filepath or datapoint_count (you already know they'll both be `None` since it's coming from `initialization_payload`. Same for TRANSFORM_SPLIT, since the result of that job will assign those attributes. I don't think you assign attributes in jobs that won't be used in the job itself, it's kind of misleading as to what the job needs (the default value for those is `None` anyway). 

2. This might seem trivial but maybe we should implement a feature of DMLJob to create a hard copy of itself (and attributes) and set the other parameters manually. Or a another DMLJob constructor that takes another DMLJob as an argument and sets the remaining arguments as the remaining attributes. For example:
```
job_two = DMLJob(
          job=job_one,
          job_type=JobTypes.JOB_INIT.name
)
 ```
Just seems repetitive to have to initialize a job from scratch if you're creating the job from another job (most of the attributes you're directly copying anyway).

Reply:
Done